src/chain/store/chain_store.rs

```diff
@@ -194,11 +196,15 @@ where
 
     /// Returns Tipset from key-value store from provided CIDs
     #[tracing::instrument(skip_all)]
-    pub fn tipset_from_keys(&self, tsk: &TipsetKeys) -> Result<Arc<Tipset>, Error> {
+    pub fn tipset_from_keys(&self, tsk: &TipsetKeys) -> Result<Option<Arc<Tipset>>, Error> {
         if tsk.cids.is_empty() {
```

Comment:
Would you happen to know if this feature is ever used? Getting the heaviest tipset when requesting an empty set of keys is odd. Maybe we can find all the places that use this and change them to call `heaviest_tipset() directly`.

Reply:
I've no idea unfortunately but I can do some investigations

Reply:
@lemmih I changed the logic to panic when `tsk.cids.is_empty()`, forest daemon seems to work fine. 

I can find some usages of this feature in RPC implementations, it allows input tipset keys to be empty.
e.g.
```rust
pub(in crate::rpc) async fn chain_get_tipset_by_height<DB>(
    data: Data<RPCState<DB>>,
    Params(params): Params<ChainGetTipsetByHeightParams>,
) -> Result<ChainGetTipsetByHeightResult, JsonRpcError>
where
    DB: Blockstore,
{
    let (height, tsk) = params;
    let ts = data.state_manager.chain_store().tipset_from_keys(&tsk)?;
    let tss = data
        .state_manager
        .chain_store()
        .chain_index
        .tipset_by_height(height, ts, ResolveNullTipset::TakeOlder)?;
    Ok((*tss).clone().into())
}
```

Reply:
Interesting, so it's probably there to match RPC behavior with Lotus.